### ISSUE TITLE
Support zero-length arrays in numba.cuda

### ISSUE THREAD

**Issue by eric-wieser** [numba/numba#6050]
As an example,
```python
@numba.cuda.jit
def foo():
    x = numba.cuda.local.array(shape=(2, 0), dtype=numba.int64)

foo()
```
gives `ValueError: array length <= 0`.

Currently these lines contain a workaround because of this issue:

https://github.com/numba/numba/blob/3399d618b334c75a7c5c1ab85ec02769fb3b4c3b/numba/cuda/cudadrv/devicearray.py#L432-L455

**Comment by gmarkall** [comment]
Thanks for the report / request!

**Comment by gmarkall** [comment]
See also #5234

**Comment by eric-wieser** [comment]
#5234 is about arrays with shape `()` (`len(shape) == 0`), this issue is about arrays with shape `(0,)`, `(0, n)` etc (`0 in shape`).

**Comment by gmarkall** [comment]
Yes, I don't think they're duplicates, I was just trying to gather together a few shape-related issues in case someone is looking at fixing up things in these areas.

**Comment by eric-wieser** [comment]
I think the other one is already fixed :)

**Comment by JuanFML** [comment]
Hello everyone! I am new here and I wanted to see if I could work on this as my first issue?

**Comment by AruparnaMaity** [comment]
Hi, @eric-wieser , @gmarkall  is this issue still open? Can I contribute in any way?

**Comment by gmarkall** [comment]
@AruparnaMaity If you can see a way to fix it, you could open a PR with the fix, if you'd like - note that as per https://numba.discourse.group/t/rfc-moving-the-cuda-target-to-a-new-package-maintained-by-nvidia/2628 new PRs should go to https://github.com/nvidia/numba-cuda

Many thanks in advance!
